src/main/scala/with_monads/Writer.scala

```diff
@@ -1,19 +1,22 @@
 package with_monads
 
-trait WriterOps[A] {
-  def value(value: A): Writer[A]
-  def tell(entries: List[String]): Writer[A]
+import cats.Semigroup
+import cats.kernel.Monoid
+
+trait WriterOps[L, A] {
+  def value(value: A): Writer[L, A]
+  def tell(entries: L): Writer[L, A]
 }
 
-case class Writer[A](log: List[String], value: A) extends WriterOps[A] { 
-  def value(newValue: A): Writer[A]          = Writer(log, newValue)
-  def tell(entries: List[String]): Writer[A] = Writer(log ++ entries, value)
+case class Writer[L: Semigroup, A](log: L , value: A) extends WriterOps[L, A] { 
+  def value(newValue: A): Writer[L, A] = Writer(log, newValue)
+  def tell(entries: L): Writer[L, A]   = Writer(Semigroup.combine(log, entries), value)
 }
 
 object Writer {
-  implicit val monadForWriter: Monad[Writer] = new Monad[Writer] {
-    def pure[A](a: A): Writer[A] = Writer(List.empty, a)
-    def flatMap[A, B](fa: Writer[A])(g: A => Writer[B]): Writer[B] = 
+  implicit def monadForWriter[L: Semigroup](implicit ev: Monoid[L]): Monad[Writer[L, *]] = new Monad[Writer[L, *]] {
```

Comment:
At this line, you do not need both `Semigroup` and `Monoid`, because `Monoid` inherits from `Semigroup`. When you remove the `Semigroup` bound on `L`. `L` will still need to be a `Monoid` because you are requiring an implicit argument of type `Monoid[L]`.